subsys/net/ip/tcp2.c

```diff
@@ -1117,6 +1120,79 @@ static void tcp_fin_timeout(struct k_work *work)
 	net_context_unref(conn->context);
 }
 
+#if IS_ENABLED(CONFIG_NET_TCP_KEEPALIVE)
+static void tcp_send_keepalive_probe(struct k_work *work)
+{
+	struct tcp *conn = CONTAINER_OF(work, struct tcp, keepalive_timer);
+
+	struct net_pkt *pkt;
+	int ret = 0;
+
+	if(!(net_context_get_state(conn->context) & NET_CONTEXT_CONNECTED))
+	{
+		NET_WARN("tcp connection not established");
```

Comment:
I would just use `NET_DBG()` here, the WARN is a bit too much to use.
Also as there can be multiple sockets that enable keepalive, it would be useful to add the conn pointer to the prints, so something like this
```
NET_DBG("[%p] tcp connection not established", conn);
```
Could you add the conn pointer print to all the debug prints you are adding in this PR